### PR TITLE
Update gateways (auth and regular) to support uploads

### PR DESCRIPTION
**Pull request by synbot**
## Update gateways (auth and regular) to support uploads

As done in Lively: 
https://github.com/synapsestudios/lively/blob/143a7af43d507b860ac30c4590fef778010935d1/application/client.js

### Acceptance Criteria
1. Can upload using synapse-common gateways

### Tasks
- Do what we did in Lively's `client.js`

### Additional Notes
- {{list additional notes here, remove if not applicable}}

### CODE REVIEW

**Comment by baohx2000** [comment]
@zpchavez have you tested this?

**Comment by zpchavez** [comment]
I'm using it currently on the branch I'm working on. I've successfully made upload requests as well as DELETE and non-upload POST requests.

File: http/gateway.js [hunk 1]
```diff
@@ -68,17 +75,75 @@ var HttpGateway = Extendable.extend({
...
+    convertArrayBufferToString : function(arrayBuffer)
+    {
+        var buffer;
+
+        buffer = new Buffer(new Uint8Array(arrayBuffer));
+
+        return buffer.toString('binary');
+    },
+
+    getBodyPrefixForFileUpload : function(file, boundaryKey)
```
Comment (paulstatezny) [inline http/gateway.js]: Could we add a brief docblock explaining this method's purpose?

**Comment by baohx2000** [comment]
:pray:

**Comment by baohx2000** [comment]
not QA testable anywhere yet

**Comment by baohx2000** [comment]
:+1:

**Pull request merged**